src/workloads/execution/BackgroundIndexConstruction.yml

```diff
@@ -0,0 +1,156 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/server-execution"
+Description: |
+  This workload tests the write performance impact of creating indexes on a very large collection.
+  It synthetically lowers the number of concurrent operations allowed to represent a saturated
+  server.
+
+Keywords:
+- stress
+- indexes
+- InsertRemove
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 1024
+      socketTimeoutMS: 3600000  # = 1 hour
+
+Actors:
+# Phase 1: Insert enough data to ensure than an index build on each field spills to disk with a
+# memory limit of 100MB.
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000000
+    BatchSize: 1000
+    Document:
+      randomInt: {^RandomInt: {min: -10000000, max: 10000000}}
```

Comment:
Does it matter to this workload that these two indexes should be fairly distributed? That is, I expect there will be few keys with more than 1 entry. Would anything interesting come out from having an indexed field with limited values, or with hot values?

Reply:
It shouldn't matter that much as indexes aren't affected by hot keys. In this case we are only interested in the background aspect of the index build occurring simultaneously with a write workload